### DISCUSSION TITLE
Unable to clone other repositories in Codespaces

### DISCUSSION THREAD

**Discussion by aboo** [community/community#36228] [Codespaces] [answered]
I have enabled codespaces for a repository which is part of an organisation. The code depends on another repository from the same organisation. Meaning it needs to be able to clone the second repository in order to be able to function.

I've read the followings:
- [Managing access to other repositories within your codespace](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces)
- [Codespaces for multi-repository and monorepo scenarios](https://github.blog/2022-04-20-codespaces-multi-repository-monorepo-scenarios/)

and I ended up with the following `devcontainer.json` file

```json
{
  "build": {
    "dockerfile": "Dockerfile"
  },
  "name": "Flutter",
  "extensions": ["Dart-Code.dart-code", "Dart-Code.flutter","auchenberg.vscode-browser-preview"],
  "forwardPorts": [3000],
  "customizations": {
      "codespaces": {
        "repositories": {
          "org/repo": {
            "permissions": "write-all"
          }
        }
      }
    }
}
```

Still when I try to clone the container using either of the commands below it won't work:

```bash
git clone https://github.com/org/repo.git
Cloning into 'repo'...
remote: Write access to repository not granted.
fatal: unable to access 'https://github.com/org/repo.git/': The requested URL returned error: 403
```
or

```bash
git clone git@github.com:org/repo.git
Cloning into 'repo'...
The authenticity of host 'github.com (20.205.243.166)' can't be established.
ECDSA key fingerprint is SHA256:p2QAMXNIC1TJYWeIOttrVc98/R1BUFWu3/LiyKgUfQM.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added 'github.com,20.205.243.166' (ECDSA) to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

If I add an SSH key just like a normal machine everything works fine. But that cannot happen every time the container starts.

I've tried:

- giving specific permissions instead of `write-all`
- defining permission for `org/*`

I was wondering if anyone has any clue on what could possibly be the issue. Cheers.

**Comment by aboo** [comment, marked as answer]
The problem is solved.

I had to delete the container and recreate it again so that it asks for permission authorization. 

Not sure if there is anyway that this can be done without deleting the container.

Also the SSH mode doesn't work and I have to use the HTTPS mode.

**Reply by brosand** [reply]
Hi @aboo, I am still having this issue even after making a new codespace, was there anything else you had to do?

**Reply by aboo** [reply]
Hi @brosand. No I didn't do anything else other than what I mentioned here.

**Reply by cox-michael** [reply]
Can you explain how you got it to ask for permission authorization? I created a new one and it never prompted for that. I have a similar json file as you, but mine has `username/*` instead of `org/repo`.

**Reply by cox-michael** [reply]
Figured it out. I hadn't properly setup the `devcontainer` files ([Add a dev container configuration to your repository](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/setting-up-your-project-for-codespaces)). After I did that I committed the changes and had to start a new codespace. Rebuilding and even Full rebuild didn't add the permissions. 
Thanks for sharing what you learned!

**Reply by konscodes** [reply]
Thank you for the answer! ⭐️ spent a lot of time here troubleshooting this issue. 
In the end, I created a [new repo](https://github.com/konscodes/playground.git) for the `.devcontainer` folder and its config with permissions to `write-all` for `username/*`

**Reply by euLita** [reply]
@aboo 

> hi,
and how did you achieve this feat? Please colleague, I am facing a problem of this nature.

-
        remote: HTTP Basic: Access denied. The provided password or token is incorrect or your account has 2FA enabled and you must use a personal access token instead of a password. See https://gitlab.com/help/topics/git/troubleshooting_git#error-on-git-fetch-http-basic-access-denied
-
        fatal: Authentication failed for 'https://gitlab.com/
        
        
   thanks so much in advence @aboo

**Comment by ossanna16** [comment]
Hey there @aboo and thank you for asking a good question! I'm glad you found our community 🙂

**Reply by ghost** [reply]
Still not working for me! Tried GitHub hosted and local codespace. 

I have

```
	"customizations": {
		"codespaces": {
			"repositories": {
			  "ansys/mathbox-kernel": {
				"permissions": "write-all"
			  }
			}
		},
		"vscode": {
			"extensions": [
				"ms-azuretools.vscode-docker"
			]
		}
	}
```

rebuilt from scratch several times. Always get:

```
Cloning into 'mathbox-kernel'...
remote: Write access to repository not granted.
fatal: unable to access 'https://github.com/org/repo.git/': The requested URL returned error: 403
```

When I use my exlicit access tokens in the form of:
`https://username:token@github.com/org/repo.git`

**Reply by cyborgdennett** [reply]
I have the same problem! is it fixed?

**Reply by pault-tech** [reply]
Also got this working with above configuration.  Had to delete the codespace and recreate to get the permission authorization dialog (full rebuild container wasn't sufficient)

Also seeing that only the https (not ssh) git protocol is working (perhaps ssh port to github.com is blocked)

**Comment by Tectract** [comment]
I'm trying to clone a repo within my gitthub codepace jupyter terminal and it's failling. What do I do to gain access? I need only read access not write access. The docs are super super unclear about how to do this BASIC task!
